command/agent/job_endpoint.go

```diff
@@ -401,6 +406,91 @@ func (s *HTTPServer) jobRunAction(resp http.ResponseWriter, req *http.Request, j
 	return s.execStream(conn, &args)
 }
 
+func (s *HTTPServer) jobTagVersion(resp http.ResponseWriter, req *http.Request, jobID string, name string) (interface{}, error) {
+	switch req.Method {
+
+	// TODO: Request for reviewers: I am splitting by method here, but you'll notice both
+	// methods now call the same RPC method, which later splits them based on presence of args.Tag.
+	// So far the benefit of a method-based split here is that I can more easily create a slimmed-down
+	// TagVersionRequest for the RPC method, but I'm not sure if that's enough to justify the split.
+	// ...is this the best and most obvious place to set the .Tag or not?
```

Comment:
These HTTP endpoints are definitely the best place to set the `.Tag`, because this is the first point where we go from 2 APIs (PUT and DELETE) to a single API (the RPC handler). 

Splitting the work into two methods seems fine to me. But if you didn't want to do that you could have each case here construct the args and then have the RPC call at the end of this `jobTagVersion` function after the switch statement. It makes it a little easier to read and to write tight little tests if you keep it the way you've got it here IMO, but it's a matter of taste.